File: server/models/chef.js
const mongoose = require('mongoose');

const Schema = mongoose.Schema;

const ChefSchema = new Schema({
    ChefID: {
        type: Schema.Types.ObjectId,
        ref: 'User',
        required: true
    },
    Bio: {
        type: String,
        required: true
    },



    Experience: {
        type: Number,
        required: true
    },
    Specialties: {
        type: [String],
        required: true
    },
    Rating: {
        type: Number,
        default: 0
    },
    AvailabilityStatus: {
        type: String,
        enum: ['Available', 'Unavailable'],
        default: 'Unavailable'
    },
    PricingPerHour: {
        type: Number,
        required: true
    },
    ProfileVerificationStatus: {
        type: String,
        enum: ['Pending', 'Verified', 'Rejected'],
        default: 'Pending'
    },
    VerifiedBadge: {
        type: Boolean,
        default: false
    },
    BackgroundImage:{
        type: String,
        required: false,
    },
    GigImage:{
        type: String,
        required: false,
    }



});

module.exports = mongoose.model('Chef', ChefSchema);

File: lib/pages/cuisine/models/chef.dart
class Chef {
  final String id;
  final String bio;
  final String username;
  final int experience;
  final List<String> specialties;
  final double rating;
  final String availabilityStatus;
  final double pricingPerHour;
  final String profileVerificationStatus;
  final bool verifiedBadge;
  final String profileImage;
  final String name;
  final String email;
  final String password;
  final String address;
  final String type;
  final bool isVerified;
  final String backgroundImage;
  final String gigImage;

  Chef({
    required this.id,
    required this.bio,
    required this.experience,
    required this.specialties,
    required this.rating,
    required this.availabilityStatus,
    required this.pricingPerHour,
    required this.profileVerificationStatus,
    required this.verifiedBadge,
    required this.profileImage,
    required this.name,
    required this.email,
    required this.password,
    required this.address,
    required this.type,
    required this.isVerified,
    this.backgroundImage = "",
    this.gigImage = "",
    this.username = "",
  });

  factory Chef.fromJson(Map<String, dynamic> json) {
    return Chef(
      id: json['_id'],
      bio: json['Bio'],
      experience: json['Experience'],
      specialties: List<String>.from(json['Specialties']),
      rating: json['Rating'].toDouble(),
      availabilityStatus: json['AvailabilityStatus'],
      pricingPerHour: json['PricingPerHour'].toDouble(),
      profileVerificationStatus: json['ProfileVerificationStatus'],
      verifiedBadge: json['VerifiedBadge'],
      profileImage: json['profileImage'],
      name: json['name'],
      email: json['email'],
      password: json['password'],
      address: json['address'],
      type: json['type'],
      isVerified: json['isVerified'],
      backgroundImage: json['backgroundImage'] ?? "",
      gigImage: json['gigImage'] ?? "",
      username: json['username'] ?? "",
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'Bio': bio,
      'Experience': experience,
      'Specialties': specialties,
      'Rating': rating,
      'AvailabilityStatus': availabilityStatus,
      'PricingPerHour': pricingPerHour,
      'ProfileVerificationStatus': profileVerificationStatus,
      'VerifiedBadge': verifiedBadge,
      'profileImage': profileImage,
      'name': name,
      'email': email,
      'password': password,
      'address': address,
      'type': type,
      'isVerified': isVerified,
      'gigImage': gigImage,
      'backgroundImage': backgroundImage,
      'username': username,
    };
  }
}
